src/download_target.cpp

```diff
@@ -59,10 +55,12 @@ namespace powerloader
             const std::string mirror_url = uh.url_without_path();
             const fs::path dst = destination_path.empty() ? fs::path{ rsplit(path, "/", 1).back() }
                                                           : destination_path;
+            spdlog::warn("SETTING MIRRORR >>>> {}, {}", host, mirror_url);
 
-            ctx.mirror_map.create_unique_mirror<Mirror>(host, ctx, mirror_url);
+            ctx.mirror_map.create_unique_mirror<Mirror>(mirror_url, ctx, mirror_url);
```

Comment:
Just to be sure: the original code in the `cli`'s main.cpp was using the variable `host` as key, which is why it was like that before. It doesnt change much of the meaning here but I'm not sure if it's ok.